tests/keyboard/scancode/test_unicode_to_scancode.cpp

```diff
@@ -5,297 +5,141 @@
 namespace
 {
     using namespace scancode;
-
-    Scancode16bits_t get_and_pop_elem(ScancodeSeq_t& scancode_seq)
-    {
-        Scancode16bits_t scancode = scancode_seq.front();
-
-        scancode_seq.pop();
-        return scancode;
-    }
-
-
+    
+    
     class FrenchLocaleFixture
     {
       protected :
-        static constexpr int _LCID = 0x0000040c;
+        static constexpr int LCID = 0x0000040c;
     };
+
+    
+#define UNI_TO_SCAN_TEST_INIT(unicode)                                  \
+    ScancodeSeq scancode_seq = unicode_to_scancode(LCID, unicode);      \
+    array_view<Scancode16bits> scancodes = scancode_seq.scancodes();    \
+
+#define UNI_TO_SCAN_TEST_EXPECT(expected_size, ...)                 \
+    RED_REQUIRE_EQUAL(scancodes.size(), expected_size);             \
+                                                                    \
+    constexpr std::uint16_t EXPECTED_SEQ[] = {__VA_ARGS__};         \
+                                                                    \
+    RED_CHECK_EQUAL(scancodes, make_array_view(EXPECTED_SEQ));      \
+
+    
+    // Need because macro __VA_ARGS__ cannot take 0 arguments
+#define UNI_TO_SCAN_TEST_EMPTY_EXPECT()         \
+    RED_CHECK(scancodes.empty());               \
+
 }
 
 
 RED_FIXTURE_TEST_CASE(unicode_to_scancode__simple_unicode_no_key_combination,
                       FrenchLocaleFixture)
 {
     {
-        constexpr std::uint16_t UNICODE = 0x0061; // 'a'
-        ScancodeSeq_t scancode_seq = unicode_to_scancode(_LCID, UNICODE);
-
-        RED_REQUIRE_EQUAL(scancode_seq.size(), 1);
-
-        
-        Scancode16bits_t pressed_scancode = get_and_pop_elem(scancode_seq);
-        
-        RED_CHECK_EQUAL(pressed_scancode, 0x0010); // 'A' scancode
+        UNI_TO_SCAN_TEST_INIT(0x0061); // 'a'
+        UNI_TO_SCAN_TEST_EXPECT(1, 0x0010); // ['A'] scancodes
```

Comment:
Separating the function call and comparison does not display the input values in the error message, information is lost. In addition, creating hidden variables forces you to add a scope which makes reading more difficult.

Reply:
What do you mean by "error message" ? At compile or runtime? I dont' see exactly what message you mean.

For your last sentence, you're right : it's the downside to use macro when it concerns variable creation :/ 
I used it mainly for factorize the code and because I already used scopes for each sub test.

So, do you suggest to come back to the previous format on the previous commit without to use  UNI_TO_SCAN_TEST_INIT and UNI_TO_SCAN_TEST_EXPECT macros?